### DISCUSSION TITLE
Trying to add voice file cannot figure it out

### DISCUSSION THREAD

**Discussion by rimwulf** [espeak-ng/espeak-ng#1384] [Q&A]
I have no idea what espeak-data/voices use it whether it is unfortunately I don't speak programmer can you elaborate, please?
Ref link:
http://espeak.sourceforge.net/voices.html

**Comment by jaacoppi** [comment]
The link you provided is for espeak. We are espeak-NG.

Read [docs/voices.md]( https://github.com/espeak-ng/espeak-ng/blob/master/docs/voices.md) and see examples in [espeak-ng-data/voices/!v](https://github.com/espeak-ng/espeak-ng/tree/master/espeak-ng-data/voices/!v). Then ask a more specific question.

**Comment by SirLaffalot** [comment]
The current android version of this app (on F-Droid) cannot be used with Android 11.
I'm on LineageOS 18.1, microG proxied Goog services.  eSpeak-NG does not ask permission for any files on the sdcard and therefore cannot load any languages/voices because it cannot read the sdcard. See screenshot attached.
The dropdown box is empty even though there is a zip file on the emulated /sdcard.
![photo_2022-09-17_14-52-25](https://user-images.githubusercontent.com/9166635/190876062-73b7a178-370d-4944-948e-ff89956100c6.jpg)

**Comment by SirLaffalot** [comment]
UPDATE ON ABOVE:

Nowhere is this procedure explained.

The solution seems to be to install a helper app from F-Droid called TTS Util by Dane Finlay. Go to the settings on this app and use first option to pick eSpeak. Then use 2nd option, TTS Voice, which causes eSpeak to unpack the voices. Now eSpeak will show 131 voices.

**Comment by rimwulf** [comment]
> The link you provided is for espeak. We are espeak-NG.
> 
> Read [docs/voices.md](https://github.com/espeak-ng/espeak-ng/blob/master/docs/voices.md) and see examples in [espeak-ng-data/voices/!v](https://github.com/espeak-ng/espeak-ng/tree/master/espeak-ng-data/voices/!v). Then ask a more specific question.

The info on the link I provided and you provided are nearly identical save for a few differences. so, my question would still stand. The problem is that everything in under-explained and tried 4 different forums this including the one within the speak-ng app itself AWA YouTube no dice. Nobody could figure out what to do on the app to add voices.

**Comment by rimwulf** [comment]
> UPDATE ON ABOVE:
> 
> Nowhere is this procedure explained.
> 
> The solution seems to be to install a helper app from F-Droid called TTS Util by Dane Finlay. Go to the settings on this app and use first option to pick eSpeak. Then use 2nd option, TTS Voice, which causes eSpeak to unpack the voices. Now eSpeak will show 131 voices.

Why it requires an SD card is beyond me. This feature seems too be really old where you could easily put local with modern Android. Unfortunately I'm using Android 12 so I'm not sure if this will work but I'll try it.

**Reply by SirLaffalot** [reply]
It doesn't require an SD card. (If your phone doesn't have an sd-card, then Android emulates it anyway.) My procedure didn't mention an SD, so forget that part and just do what I said. It works. And even though it says eSpeak, it means eSpeak-ng, which is a fork and updated. You'd never be able to load the original on Android 12 anyway.
